dartagnan/src/main/java/com/dat3m/dartagnan/analysis/graphRefinement/graphs/eventGraph/stat/RMWGraph.java

```diff
@@ -0,0 +1,84 @@
+package com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.stat;
+
+import com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.EventGraph;
+import com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.utils.MaterializedGraph;
+import com.dat3m.dartagnan.analysis.graphRefinement.util.GraphVisitor;
+import com.dat3m.dartagnan.program.event.rmw.RMWStore;
+import com.dat3m.dartagnan.verification.model.Edge;
+import com.dat3m.dartagnan.verification.model.EventData;
+import com.dat3m.dartagnan.verification.model.ExecutionModel;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class RMWGraph extends MaterializedGraph {
+
+    public RMWGraph() { }
+
+    @Override
+    public List<? extends EventGraph> getDependencies() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void constructFromModel(ExecutionModel model) {
+        super.constructFromModel(model);
+        populate();
+    }
+
+    @Override
+    public <TRet, TData, TContext> TRet accept(GraphVisitor<TRet, TData, TContext> visitor, TData data, TContext context) {
+        return visitor.visitBase(this, data, context);
+    }
+
+
+    /* There are four cases where the RMW-Relation is established:
+     (1) LOCK : Load -> CondJump -> Store
+     (2) RMW : RMWLoad -> RMWStore (completely static)
+     (3) ExclAccess : ExclLoad -> ExclStore (dependent on control flow)
+     (4) Atomic blocks: BeginAtomic -> Events* -> EndAtomic
+    */
```

Comment:
Maybe we should add some cross-reference with RelRMW to be sure that if some semantic changes is done there (like the one we did when we started using RMW for atomic blocks), we need to also do something here